src/Microsoft.AspNet.Hosting.Server.Abstractions/IHttpApplication.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Features;
+
+namespace Microsoft.AspNet.Hosting.Server
+{
+    /// <summary>
+    /// Represents an HttpApplication.
+    /// </summary>
+    public interface IHttpApplication
+    {
+        /// <summary>
+        /// Create an HttpContext given a collection of HTTP features.
+        /// </summary>
+        /// <param name="contextFeatures">A collection of HTTP features to be used for creating the HttpContext.</param>
+        /// <returns>The created HttpContext.</returns>
+        HttpContext CreateHttpContext(IFeatureCollection contextFeatures);
```

Comment:
HttpContext in all 3 places should be object.

@davidfowl What is the reason it couldn't be a `<TContext>` on the interface and Start method?